core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -57,6 +57,15 @@ object DataMirror {
     */
   def hasProbeTypeModifier(x: Data): Boolean = x.probeInfo.nonEmpty
 
+  /** Return the optional layer color of a `Data`.
+    * @param x the `Data` to examine
+    * @return a `Some[Layer]` if the data has a layer color, `None` otherwise
+    */
+  def getLayerColor(x: Data): Option[layer.Layer] = x.probeInfo match {
+    case None                           => None
+    case Some(Data.ProbeInfo(_, color)) => color
+  }
```

Comment:
```suggestion
  def getLayerColor(x: Data): Option[layer.Layer] =
    x.probeInfo.collect { case Data.ProbeInfo(_, color) => color }
```
`case None => None` always suggests a `.collect` to me. But feel free to ignore 🙂

Reply:
This is a bit cleaner. Nice suggestion. Changed in https://github.com/chipsalliance/chisel/pull/3765/commits/2f1284f52e836357f845225483cbcc6a9267083a.